src/stormpath.social-login.facebook.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+angular.module('stormpath.facebookLogin', [])
+.provider('$facebookLogin', function() {
+  function loadFacebookSDK() {
+    var id = 'facebook-jssdk';
+    var firstJs = document.getElementsByTagName('script')[0];
+    var js;
+
+    if (document.getElementById(id)) {
+      return;
+    }
+
+    js = document.createElement('script');
+    js.id = id;
+    js.src = '//connect.facebook.net/en_US/sdk.js';
+
+    firstJs.parentNode.insertBefore(js, firstJs);
+  }
+
+  function loginCallback(deferred, response) {
+    var data;
+
+    switch (response.status) {
+      case 'connected':
+        deferred.resolve({
+          providerData: {
+              providerId: 'facebook',
+              accessToken: response.authResponse.accessToken
+          }
+        });
+        break;
+
+      case 'not_authorized':
+        deferred.reject(new Error('Please log into this app'));
+        break;
+
+      default:
+        deferred.reject(new Error('Please log into Facebook.'));
+    }
+  }
+
+  this.$get = ['$window', '$q', function facebookLoginFactory($window, $q) {
+    var clientId;
+
+    function FacebookLoginService() {
+      return this;
```

Comment:
This is not needed. Also, a "constructor" should never return anything except itself (default).